src/DynamoCore/Core/DynamoMigrator.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Xml.Serialization;
+using Dynamo.Models;
+using ProtoCore.Lang;
+
+namespace Dynamo.Core
+{
+    internal abstract class DynamoMigratorBase
```

Comment:
Please supply extensive documentation for at least the base class `DynamoMigratorBase` and its public facing APIs.